### DISCUSSION TITLE
Looping replacement support

### DISCUSSION THREAD

**Discussion by davegraham92120** [mrwatson-de/fmWorkMate#65] [Q&A] [answered]
@mrwatson-de I'd like to use fmTextMultiplier to generate a script that performs some operations on a number of different fields. fmTextMultiplier can generate the inner steps but I'm not seeing a way to generate the entire script. I'm wondering if there is a way to accomplish this or if we should consider adding support for loops within the text ([example](https://handlebarsjs.com/examples/builtin-helper-each-block.html)).

I want to generate a script like so: 

some XML preamble content
multiplied content block
some XML trailing content

**Comment by mrwatson-de** [comment, marked as answer]
Hi Dave,

Great that you are using fmTextMultiplier ! And congratulations, I think you are the first to interact with my GitHub repo! 👍

While fmTextMultiplier is an extremely useful little tool for multiplying code up for day to day manual coding, I'm not sure it is the easiest tool to use for your use case.

Maybe the tool could be kitted out with a pre-text and a post-text, but that seems a little static/restricted compared to what I imagine one might want. 

The example you link to is a fully fledged programming language, and we may be able to get such results more quickly with Geist interactive/Proof-Geist's generator.

I'm afraid I don't have a link since I've not used it myself (plus am currently in the **very** rare situation of being in the bath 🛁😆), but I've understood it's got scripting functions for building up code using a FileMaker script. That might be more suitable for what you want.

I originally imagined my various tools to be like components on a circuit board, that you can link up to create more complex logic, however in the decade I've had them no technique has manifested to be able to do that in a flexible and easy way.

I hope this helps, and am open to all ideas and further discussions!

**Comment by davegraham92120** [comment]
I figured that was the case, but wanted to check before I created a specific feature in Generator for this purpose. I also think that we should consider native Handlebars support in fmTextMultiplier since it's so robust. I'm going to start tinkering...

**Comment by mrwatson-de** [comment]
😂 nice!

I'm intrigued to see what comes out of that.

I've been looking at XQuery over the years.

For simple stuff it might even be possible to extend the Insert Pseudocode command in fmAutoMate.

I'm also trying to get my teeth into domain specific languages - but lacking the time.

What about embedding (handlebar-) logic into (script and/or calculation) comments of real FileMaker Scripts, so that you can make a 'template script' which then gets interpreted, multiplying and processing the code as defined in the mark up?

Something like this:

```
# Start Code here
   :
# @{ Multiply ( FieldA/FieldB/FieldC )
Set Field [ MyTable::FieldA ; "" ]
# @}
  :
# End Code here
```

**Reply by davegraham92120** [reply]
I can see the motivation to embed Handlebars code; at least at a rudimentary level. It would allow you to revise code in FileMaker and easily copy the refactored code out, however, I don't think the extra work is going to be worth it in the end.

If you have to write the Handlebars code anyway, I'm guessing that it's more efficient to use a code editor with Handlebars support rather than embed it into FileMaker. Handlebars code embedded in comments would have to be parsed, adding additional overhead, and I'm guessing that there are some Handlebars functions that would be near impossible to replicate in script comments.

**Comment by mrwatson-de** [comment]
P.S. I've just posted a bug report on the community where the demo file includes a custom function G_TX_Format_Text that i made for (simply) formatting text (in different output environments) - also a little inspired from handlebars (but far from it!).  Just search the CF name.

**Comment by mrwatson-de** [comment]
Hi,

yes, but…

isn't it a question of 
either implementing handlebars code in FileMaker
or implementing FileMaker XML in Handlebars ?

I'm not sure which is the easier task ! :D
